handshake/crypto_setup_client_test.go

```diff
@@ -640,13 +640,15 @@ var _ = Describe("Client Crypto Setup", func() {
 		})
 
 		It("tries to escalate the crypto after receiving a diversification nonce", func(done Done) {
-			go cs.HandleCryptoStream()
-			time.Sleep(50 * time.Millisecond) // wait for the first maybeUpgradeCrypto to finish
+			go func() {
+				defer GinkgoRecover()
+				cs.HandleCryptoStream()
+				Fail("HandleCryptoStream should not have returned")
```

Comment:
I'll do the chan solution for unblocking the read. That should take care of all leaking `go HandleCryptoStream` routines.